xapian-letor/api/featurevector.cc

```diff
@@ -48,14 +48,33 @@ using namespace std;
 
 using namespace Xapian;
 
-FeatureVector::FeatureVector() {
+class FeatureVector::Feature_internal : public Xapian::Internal::intrusive_base
```

Comment:
You can just call this `FeatureVector::Internal` (similar to `Document::Internal`).